crates/uv/tests/it/pip_compile.rs

```diff
@@ -15230,6 +15230,274 @@ fn respect_index_preference() -> Result<()> {
     Ok(())
 }
 
+#[test]
+fn dependency_group() -> Result<()> {
+    fn new_context() -> Result<TestContext> {
+        let context = TestContext::new("3.12");
+
+        let pyproject_toml = context.temp_dir.child("pyproject.toml");
+        pyproject_toml.write_str(
+            r#"
+            [project]
+            name = "project"
+            version = "0.1.0"
+            requires-python = ">=3.12"
+            dependencies = ["typing-extensions"]
+            [dependency-groups]
+            foo = ["sortedcontainers"]
+            bar = ["iniconfig"]
+            dev = ["sniffio"]
+            "#,
+        )?;
+
+        Ok(context)
+    }
+
+    let mut context;
+
+    context = new_context()?;
+    uv_snapshot!(context.filters(), context.pip_compile()
+        .arg("--group").arg("bar"), @r"
+    success: true
+    exit_code: 0
+    ----- stdout -----
+    # This file was autogenerated by uv via the following command:
+    #    uv pip compile --cache-dir [CACHE_DIR] --group bar
+    iniconfig==2.0.0
+
+    ----- stderr -----
+    Resolved 1 package in [TIME]
+    ");
+
+    context = new_context()?;
+    uv_snapshot!(context.filters(), context.pip_compile()
+        .arg("pyproject.toml")
+        .arg("--group").arg("bar"), @r"
+    success: true
+    exit_code: 0
+    ----- stdout -----
+    # This file was autogenerated by uv via the following command:
+    #    uv pip compile --cache-dir [CACHE_DIR] pyproject.toml --group bar
+    iniconfig==2.0.0
+    typing-extensions==4.10.0
+        # via project (pyproject.toml)
```

Comment:
This could be separate, but... should we be emitting annotations for this? e.g.
```suggestion
    iniconfig==2.0.0
        # via group (pyproject.toml:bar)
    typing-extensions==4.10.0
        # via project (pyproject.toml)
```

Reply:
As an aside, these would be so helpful for reading these test cases :D

Reply:
Ok I have an implementation but worth noting here that `project` is not the literal string "project" but is the actual name of the project (we just call our test projects "project" lol). So changing that to mycoolproj we currently print:

```
   # via mycoolproj (path/to/pyproject.toml)
```

Your proposal would naively be:

```
   # via mycoolproj (path/to/pyproject.toml:bar)
```

I can do that, but lmk if you want something Different.

Reply:
(ugh, we should _not_ call our test projects "project" lol)

It's not a dependency of the package, so... I'm not sure we want to say the project name.

Anyway I don't think we need to bikeshed this. Let's do something vaguely sensible and we can change it in the future? cc @charliermarsh 

Can you also cross-post whatever we land on / or prod for discussion now upstream in the `pip-tools` issue?

Reply:
Oh geez is the exact syntax of this comment load-bearing to something.

Reply:
Or is this just a "let's not needlessly deviate from the other tool's output, even for comments"?

Reply:
Left a message with them.